### PR TITLE
Fix MetricsTracker

### PR DESCRIPTION
**Pull request by chenjiasheng**
When running libritts debug configuration on 8 GPUs, one rank first consumes its data loader, and then enters the `compute_validation_loss` function and stuck on the `tot_loss.reduce(loss.device)` statement. At the same time other ranks are still training the next batch and trying to wait for this rank. Eventually, it leads to dead hanging.

In this fix, I all_gather stop_iteration status from all ranks, if any of them stops, then stop all the ranks.

I wonder why you guys dont encounter this hanging problem. Correct me if I am wrong.

@lifeiteng

### CODE REVIEW

**Comment by lifeiteng** [comment]
@chenjiasheng LGTM, but I'm not familiar with `dist`, I'll merge when verified in https://github.com/lifeiteng/vall-e/issues/86

**Review by lifeiteng — APPROVED** [review]

**Comment by chenjiasheng** [comment]
additional commit:
Fix moving mean calculation of tot_loss to prevent the loss curve from being unstable.
@lifeiteng

File: valle/bin/trainer.py [hunk 1]
```diff
@@ -683,7 +683,7 @@ def evaluate():
                     is_training=True,
                 )
             # summary stats
-            tot_loss = (tot_loss * (1 - 1 / params.reset_interval)) + loss_info
+            tot_loss = (tot_loss * (1 - 1 / params.reset_interval)) + loss_info / params.reset_interval
```
Comment (lifeiteng) [inline valle/bin/trainer.py:686]: @chenjiasheng 我觉着你可以给 icefall 提一波 PR 哈哈哈哈，commit massage 认真写下

Comment (chenjiasheng) [reply valle/bin/trainer.py:686]: 嗯，给icefall提issue了，没有提PR。公司网络推github太费劲了。

Comment (chenjiasheng) [reply valle/bin/trainer.py]: 先前提的确实有问题
loss_info / params.reset_interval的写法不对，
得改成 loss_info *  (1 / params.reset_interval)

我刚才实际跑了一下，这次应该没问题了
